tests/php/static-analysis-stubs/pwa.php

```diff
@@ -0,0 +1,280 @@
+     * @type array   $plugins    Array of plugins with configuration. The key of each plugin in the array must match the plugin's name.
+     *                              See https://developers.google.com/web/tools/workbox/guides/using-plugins#workbox_plugins.
+     * }
+     * @since 0.2
+     *
+     */
+    public function register($route, $args = array())
+    {
+    }
+    /**
+     * Gets all registered routes.
+     *
+     * @return array List of registered routes.
+     * @since 0.2
+     *
+     */
+    public function get_all()
+    {
+        return $this->routes;
+    }
+    /**
+     * Prepare caching strategy args for export to JS.
+     *
+     * @param array $strategy_args Strategy args.
+     * @return string JS IIFE which returns object for passing to registerRoute.
+     * @since 0.2
+     *
+     */
+    public static function prepare_strategy_args_for_js_export($strategy_args)
+    {
+    }
+}
+/**
+ * Interface for classes that host a registry.
+ *
+ * @since 0.2
+ */
+interface WP_Service_Worker_Registry_Aware {
+	/**
+	 * Gets the registry.
+	 *
+	 * @return WP_Service_Worker_Registry Registry instance.
+	 */
+	public function get_registry();
+}
+/**
+ * Class used to register service workers.
+ *
+ * @since 0.1
+ *
+ * @see   WP_Dependencies
+ */
+class WP_Service_Workers implements WP_Service_Worker_Registry_Aware
+{
+    /**
+     * Param for service workers.
+     *
+     * @var string
+     */
+    const QUERY_VAR = 'wp_service_worker';
+    /**
+     * Scope for front.
+     *
+     * @var int
+     */
+    const SCOPE_FRONT = 1;
+    /**
+     * Scope for admin.
+     *
+     * @var int
+     */
+    const SCOPE_ADMIN = 2;
+    /**
+     * Scope for both front and admin.
+     *
+     * @var int
+     */
+    const SCOPE_ALL = 3;
+    /**
+     * Constructor.
+     *
+     * Instantiates the service worker scripts registry.
+     */
+    public function __construct()
+    {
+    }
+    /**
+     * Gets the service worker scripts registry.
+     *
+     * @return WP_Service_Worker_Scripts Scripts registry instance.
+     */
+    public function get_registry()
+    {
+    }
+    /**
+     * Get the current scope for the service worker request.
+     *
+     * @todo We don't really need this. A simple call to is_admin() is all that is required.
+     * @return int Scope. Either SCOPE_FRONT or SCOPE_ADMIN.
+     */
+    public function get_current_scope()
+    {
+    }
+    /**
+     * Get service worker logic for scope.
+     *
+     * @see wp_service_worker_loaded()
+     */
+    public function serve_request()
+    {
+    }
+}
+/**
+ * Get service worker URL by scope.
+ *
+ * @since 0.1
+ *
+ * @param int $scope Scope for which service worker to output. Can be WP_Service_Workers::SCOPE_FRONT (default) or WP_Service_Workers::SCOPE_ADMIN.
+ * @return string Service Worker URL.
+ */
+function wp_get_service_worker_url( $scope = WP_Service_Workers::SCOPE_FRONT ) {}
```

Comment:
How is this file generated?

Reply:
Oh, via https://github.com/szepeviktor/phpstan-wordpress?

Reply:
I think by hand.

Reply:
This one was assembled by hand.
All of the stubs will currently stay at the specific version I submitted via this PR. If maintenance becomes an issue at some point, we can look at automating the generation for everything. However, I think this is quite involved and might not be worth the hassle. PHPStan will let us know anyway if something's missing.